### PR TITLE
Puppet services padlocal

### PR DESCRIPTION
**Pull request by Soumi7**
# Puppet services padlocal

References issue #890

### CODE REVIEW

**Comment by Soumi7** [comment]
Screenshots :

Screenshot from 2021-09-11 11-32-27

Screenshot from 2021-09-11 11-32-38

Screenshot from 2021-09-11 11-32-50

Screenshot from 2021-09-11 11-32-59

Screenshot from 2021-09-11 11-33-03

Screenshot from 2021-09-11 11-33-05

Screenshot from 2021-09-11 11-33-09

Screenshot from 2021-09-11 11-33-12

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]

**Review by shraddhavp — APPROVED** [review]
LGTM

File: docusaurus/docs/puppet-services/padlocal.md [hunk 1]
```diff
@@ -5,16 +5,118 @@ sidebar_label: PadLocal
...
+### Steps
+
+1. Run wechaty-puppet on a virtual machine
+2. Verify that the API is successfully registered
+3. Run local python-wechaty-getting-started
+4. Scan code login
+
+### Step 1 : wechaty-puppet-padlocal
+
+You will need a domestic public network IP+Token (please register with the administrator) to make wechaty-puppet take effect in the virtual machine part. For the trial or free plan, you can choose Alibaba Cloud or Huawei Cloud or GCP. It is a pity that Google does not have a domestic IP. , Or you can also play gcp's k8s by the way, suggest that you still use domestic resources.
```
Comment (proudofsimin) [inline docusaurus/docs/puppet-services/padlocal.md]:
```suggestion
You will need a domestic public network IP+Token (please register with the administrator) to make wechaty-puppet take effect in the virtual machine part. For the trial or free plan, you can choose Alibaba Cloud or Huawei Cloud or GCP. It is a pity that Google does not have a domestic IP. , or you can also play gcp's k8s by the way, suggest that you still use domestic resources.
```

File: docusaurus/docs/puppet-services/padlocal.md [hunk 2]
```diff
@@ -5,16 +5,118 @@ sidebar_label: PadLocal
...
+
+![Step 1-terminal](../../static/img/docs/references/puppet-services/padlocal/step-1-terminal.webp)
+
+You can telnet the virtual machine IP: 8788, simple debugging will not work.
+(Optional) You can change docker run to docker-compose by running, and the virtual machine can grow up automatically after it crashes.
+
+### Step 2 : verify hostie token
+
+Visit [this link](https://api.chatie.io/v0/hosties/puppet_padlocal_xxxxxxxx).
+To check whether your token is successfully registered, if not, please go back to step 1 to check.
```
Comment (proudofsimin) [inline docusaurus/docs/puppet-services/padlocal.md]:
```suggestion
Visit [this link](https://api.chatie.io/v0/hosties/puppet_padlocal_xxxxxxxx) to check whether your token is successfully registered, if not, please go back to step 1.
```

File: docusaurus/docs/puppet-services/padlocal.md [hunk 3]
```diff
@@ -5,16 +5,118 @@ sidebar_label: PadLocal
...
+![Architecture](../../static/img/docs/references/puppet-services/padlocal/architecture.webp)
+
+### Steps
+
+1. Run wechaty-puppet on a virtual machine
+2. Verify that the API is successfully registered
+3. Run local python-wechaty-getting-started
+4. Scan code login
+
+### Step 1 : wechaty-puppet-padlocal
```
Comment (proudofsimin) [inline docusaurus/docs/puppet-services/padlocal.md]:
```suggestion
### Step 1 : run wechaty-puppet-padlocal
```

File: docusaurus/docs/puppet-services/padlocal.md [hunk 4]
```diff
@@ -5,16 +5,118 @@ sidebar_label: PadLocal
...
+- PC (Mac is used here)
+- Virtual machine for puppet on Alibaba Cloud
+
+## Architecture
+
+![Architecture](../../static/img/docs/references/puppet-services/padlocal/architecture.webp)
+
+### Steps
+
+1. Run wechaty-puppet on a virtual machine
```
Comment (proudofsimin) [inline docusaurus/docs/puppet-services/padlocal.md]: Should it be `wechaty-puppet-padlocal`?

File: docusaurus/docs/puppet-services/padlocal.md [hunk 5]
```diff
@@ -5,16 +5,118 @@ sidebar_label: PadLocal
...
+2. Verify that the API is successfully registered
+3. Run local python-wechaty-getting-started
```
Comment (proudofsimin) [inline docusaurus/docs/puppet-services/padlocal.md]:
```suggestion
3. Run local `python-wechaty-getting-started`
```

**Review by proudofsimin — APPROVED** [review]

File: docusaurus/docs/puppet-services/padlocal.md (hunk 4 above)
Comment (Soumi7) [reply docusaurus/docs/puppet-services/padlocal.md]: Changed, thankyou.

**Comment by Soumi7** [comment]
@proudofsimin Have made the changes you requested.

**Pull request merged**